### ISSUE TITLE
`_RetryQueue started before transport._raiden_service is set` (after unclean shutdown)

### ISSUE THREAD

**Issue by konradkonrad** [raiden-network/raiden#6071]
During rc5 mainnet testing, after unclean node shutdown (terminal died), I see:
```
Welcome to Raiden, version 0.200.0rc5!
----------------------------------------------------------------------
| This is an Alpha version of experimental open source software      |
...
----------------------------------------------------------------------
The Raiden API RPC server is now running at http://localhost:5001/.

 See the Raiden documentation for all available endpoints at
 http://raiden-network.readthedocs.io/en/stable/rest_api.html
Checking if the ethereum node is synchronized
2020-04-15 14:09:47.413920 [error    ] Error processing request       [raiden.api.rest_utils] errors=Service unavailable. Try again later. status_code=<HTTPStatus.SERVICE_UNAVAILABLE: 503>
2020-04-15 14:09:47.416051 [info     ] 127.0.0.1 - - [2020-04-15 16:09:47] "GET /api/v1/pending_transfers HTTP/1.1" 503 204 0.002770 [raiden.api.rest.pywsgi] 
You have chosen the Pathfinding Service at https://pfs.rsb01.test.raiden.network.
Operator: "RSB01 TEST PFS", running version: 0.8.0, chain_id: 1.
Fees will be paid to 0xa5041aa778Bb06712d889e406336Cb3c5aa0d1cB. Each request costs 0.05 RDN.
Message from the Pathfinding Service:
"This is the PFS for rsb01.test.raiden.network"
2020-04-15 14:09:47.891535 [info     ] Using Pathfinding Service      [raiden.network.pathfinding] pfs_info=PFSInfo(url='https://pfs.rsb01.test.raiden.network', price=50000000000000000, chain_id=1, token_network_registry_address=b'/\x88\xf8&\x18\x19\xc4\xd63F\x80J\x17\xaa\x93\xc3\\\xc7\x9dy', user_deposit_address=b'\x15\xac7\x1a\xde!\xc4\xf3\x1d\xa3jL\xf9\xa0\xef5\xaf\xdez\x9f', payment_address=b'\xa5\x04\x1a\xa7x\xbb\x06q-\x88\x9e@c6\xcb<Z\xa0\xd1\xcb', message='"This is the PFS for rsb01.test.raiden.network"', operator='"RSB01 TEST PFS"', version='0.8.0', confirmed_block_number=9877440)
Raiden is running in production mode

You are connected to the 'mainnet' network and the DB path is: /home/konrad/.raiden/node_298e4ac5/netid_1/network_2f88f826/v26_log.db
2020-04-15 14:09:49.873353 [info     ] Using Matrix server            [raiden.network.transport.matrix.utils] average_rtt=0.021244666666666665 server_ident=Synapse/1.10.0 server_url=https://transport.rsb01.test.raiden.network
2020-04-15 14:09:50.268294 [info     ] Synchronizing blockchain events [raiden.raiden_service] blocks_per_second=25281586.49797669 blocks_to_sync=9877440 elapsed=0.390697 to_block=9877440
2020-04-15 14:09:50.494431 [info     ] Updating channel fees          [raiden.raiden_service] cap_mediation_fees=True channel=CanonicalIdentifier(chain_identifier=1, token_network_address=b'4\xffL\x1c\xb4\x06\x87\xd0\xb6J\xcc\xc2AT\xd1\x87&Z?=', channel_identifier=3) flat_fee=0 proportional_fee=4000 proportional_imbalance_fee=3000
2020-04-15 14:09:50.497983 [info     ] Updating channel fees          [raiden.raiden_service] cap_mediation_fees=True channel=CanonicalIdentifier(chain_identifier=1, token_network_address=b'4\xffL\x1c\xb4\x06\x87\xd0\xb6J\xcc\xc2AT\xd1\x87&Z?=', channel_identifier=4) flat_fee=0 proportional_fee=4000 proportional_imbalance_fee=3000
2020-04-15 14:09:50.501035 [info     ] Updating channel fees          [raiden.raiden_service] cap_mediation_fees=True channel=CanonicalIdentifier(chain_identifier=1, token_network_address=b'4\xffL\x1c\xb4\x06\x87\xd0\xb6J\xcc\xc2AT\xd1\x87&Z?=', channel_identifier=10) flat_fee=0 proportional_fee=4000 proportional_imbalance_fee=3000
2020-04-15 14:09:50.503399 [info     ] Updating channel fees          [raiden.raiden_service] cap_mediation_fees=True channel=CanonicalIdentifier(chain_identifier=1, token_network_address=b'4\xffL\x1c\xb4\x06\x87\xd0\xb6J\xcc\xc2AT\xd1\x87&Z?=', channel_identifier=13) flat_fee=0 proportional_fee=4000 proportional_imbalance_fee=3000
2020-04-15 14:09:50.506122 [info     ] Updating channel fees          [raiden.raiden_service] cap_mediation_fees=True channel=CanonicalIdentifier(chain_identifier=1, token_network_address=b'4\xffL\x1c\xb4\x06\x87\xd0\xb6J\xcc\xc2AT\xd1\x87&Z?=', channel_identifier=18) flat_fee=0 proportional_fee=4000 proportional_imbalance_fee=3000
2020-04-15 14:09:50.509170 [info     ] Updating channel fees          [raiden.raiden_service] cap_mediation_fees=True channel=CanonicalIdentifier(chain_identifier=1, token_network_address=b'4\xffL\x1c\xb4\x06\x87\xd0\xb6J\xcc\xc2AT\xd1\x87&Z?=', channel_identifier=19) flat_fee=0 proportional_fee=4000 proportional_imbalance_fee=3000
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 854, in gevent._greenlet.Greenlet.run
  File "raiden/network/transport/matrix/transport.py", line 295, in _run
AssertionError: _RetryQueue started before transport._raiden_service is set
2020-04-15T14:09:50Z <Greenlet "RetryQueue recipient:0x94eEE3B101A2C153DaD1022be4eAE89EC447Bdaa" at 0x7f0d72e31ef0: <bound method _RetryQueue._run of <_RetryQueue for 0x94eee3b101a2c153dad1022be4eae89ec447bdaa>>> failed with AssertionError

2020-04-15 14:09:50.515660 [error    ] Runnable subtask died!         [raiden.utils.runnable] exc=AssertionError('_RetryQueue started before transport._raiden_service is set') running=False subtask=<Greenlet "RetryQueue recipient:0x94eEE3B101A2C153DaD1022be4eAE89EC447Bdaa" at 0x7f0d72e31ef0: <bound method _RetryQueue._run of <_RetryQueue for 0x94eee3b101a2c153dad1022be4eae89ec447bdaa>>> this=<MatrixTransport id:25de5fa3-6097-41b7-9c22-614e1cba1a28>
```
- [ ] try to reproduce with rc7 
- [ ] maybe get logs from @konradkonrad
- [ ] Maybe related: understand why channel fees are updated before Matrix Transport is there

**Comment by ulope** [comment]
Ok, this is weird.

The retriers are started here: https://github.com/raiden-network/raiden/blob/0d346f3904c9f05792e64ee508e803e22755ae5c/raiden/network/transport/matrix/transport.py#L468-L471

In that method the `_raiden_service` instance variable is used in multiple places. That means it has to be set. Therefore the assert in `_RetryQueue._run()` should pass: https://github.com/raiden-network/raiden/blob/0d346f3904c9f05792e64ee508e803e22755ae5c/raiden/network/transport/matrix/transport.py#L294-L295

**Comment by Dominik1999** [comment]
- try to reproduce with rc7 [ ] 
- maybe get logs from @konradkonrad [ ] 
- understand why channel fees are updated before Matrix Transport is there [ ]

**Comment by Dominik1999** [comment]
@konradkonrad pls add db to issue

**Comment by Dominik1999** [comment]
pair it up with karl

**Comment by konradkonrad** [comment]
[db_node_298e4a.zip](https://github.com/raiden-network/raiden/files/4487836/db_node_298e4a.zip)

**Comment by fredo** [comment]
closed due to #6086
